src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -996,10 +997,17 @@ struct FixedKVLeafNode
   // children are considered stable if any of the following case is true:
   // 1. The child extent is absent in cache
   // 2. The child extent is stable
-  bool is_child_stable(uint16_t pos) const final {
+  //
+  // For reserved mappings, the return values are undefined.
+  bool is_child_stable(op_context_t<NODE_KEY> c, uint16_t pos) const final {
     auto child = this->children[pos];
-    if (is_valid_child_ptr(child)) {
+    if (is_reserved_ptr(child)) {
+      return true;
+    } else if (is_valid_child_ptr(child)) {
       ceph_assert(child->is_logical());
+      ceph_assert(
+	child->is_pending_in_trans(c.trans.get_trans_id())
+	|| this->is_stable_written());
```

Comment:
Looking at `FixedKVLeafNode::get_logical_child()`, I'm not following why it isn't calling `get_extent_viewable_by_trans()`, I suppose they are almost identical?

And seems `is_child_stable()` will still return true if the parent is stable, and the child is mutated in the transaction, right?

Reply:
> And seems `is_child_stable()` will still return true if the parent is stable, and the child is mutated in the transaction, right?

Only mutated and under-io children will be considered stable.


> Looking at `FixedKVLeafNode::get_logical_child()`, I'm not following why it isn't calling `get_extent_viewable_by_trans()`, I suppose they are almost identical?

Because calling `get_extent_viewable_by_trans()` would make `LBAMapping::is_stable()` returning a future, which seems not necessary as it's just a state probe not an actual extent access.

Reply:
> Because calling get_extent_viewable_by_trans() would make LBAMapping::is_stable() returning a future.

Sorry, I meant -- why it isn't calling `return c.cache.is_viewable_extent_stable(c.trans, child)`?

>> And seems is_child_stable() will still return true if the parent is stable, and the child is mutated in the transaction, right?
> Only mutated and under-io children will be considered stable.

I meant if the parent is stable, and the child in transaction is mutation-pending (not under-io, the transaction isn't yet submitted), the child pin with `is_child_stable()` should return false. My understanding that here is checking the stable version of the child so it will return true in this case.

Reply:
> I meant if the parent is stable, and the child in transaction is mutation-pending (not under-io, the transaction isn't yet submitted), the child pin with `is_child_stable()` should return false. My understanding that here is checking the stable version of the child so it will return true in this case.

No, it will return false, `CachedExtent::is_stable()` returns false for non-under-io mutation-pending extents.

> Sorry, I meant -- why it isn't calling `return c.cache.is_viewable_extent_stable(c.trans, child)`?

Ah, yes, I forgot it, will change it.